### ISSUE TITLE
Conversion of Nothing literal to Date? as operand of ternary If expression incorrectly reported as unnecessary

### ISSUE THREAD

**Issue by AnthonyDGreen** [dotnet/roslyn#5685]
```VB.NET
Module Module1
    Sub Main()
        Dim d As Date? = If(True, DirectCast(Nothing, Date?), CDate(""))
    End Sub
End Module
```

The DirectCast above is faded. This also affects CType. Removing the conversion would cause the Nothing literal to be interpreted as #1/1/0001 00:00:00# rather than as a null nullable Date.
